flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -679,6 +720,26 @@ ANY, and(logical(LogicalTypeRoot.BOOLEAN), LITERAL)
     public static final BuiltInFunctionDefinition DISTINCT =
             BuiltInFunctionDefinition.newBuilder()
                     .name("distinct")
+                    .callSyntax(
+                            (sqlName, operands) -> {
+                                final CallExpression callExpression =
+                                        (CallExpression) operands.get(0);
+                                if (callExpression.getFunctionDefinition()
```

Comment:
Can you add some comments here? This looks not straightforward compared to the other functions.